src/bio/bio_xstream.c

```diff
@@ -1474,23 +1568,32 @@ bio_xsctxt_alloc(struct bio_xs_context **pctxt, int tgt_id, bool self_polling)
 
 	/* Initialize per-xstream blobstore context */
 	for (st = SMD_DEV_TYPE_DATA; st < SMD_DEV_TYPE_MAX; st++) {
-		/* only data blob now */
-		if (st != SMD_DEV_TYPE_DATA)
+		/* No Data blobstore for sys xstream */
+		if (st == SMD_DEV_TYPE_DATA && tgt_id == BIO_SYS_TGT_ID)
 			continue;
+
+		/* No Meta/WAL blobstore if Metadata on SSD is not configured */
+		if (st != SMD_DEV_TYPE_DATA && !bio_is_meta_on_ssd_configured())
+			break;
+
 		rc = init_xs_blobstore_ctxt(ctxt, tgt_id, st);
 		if (rc)
 			goto out;
 	}
 
+	if (tgt_id == BIO_SYS_TGT_ID) {
```

Comment:
Minor: It's the init thread being responsible for initializing/finalizing SMD, this could be revised when we make patch to support standalone VOS.

I think the standalone VOS support will be similar to sys xstream support, the only difference is that for sys xstream, we don't assign data blobstore, but for standalone VOS, we should assign blobstores according to the roles passed by bdev name.

Reply:
Sure